lisa/transformers/package_installer.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from dataclasses import dataclass, field
+from pathlib import PurePath
+from typing import Any, Dict, List, Type
+
+from dataclasses_json import dataclass_json
+
+from lisa import schema
+from lisa.operating_system import RPMDistro
+from lisa.tools import Rpm
+from lisa.transformers.deployment_transformer import (
+    DeploymentTransformer,
+    DeploymentTransformerSchema,
+)
+from lisa.util import UnsupportedDistroException, field_metadata
+
+
+@dataclass_json()
+@dataclass
+class PackageInstallerSchema(DeploymentTransformerSchema):
+    ...
+
+
+class PackageInstaller(DeploymentTransformer):
+    @classmethod
+    def type_name(cls) -> str:
+        return "package_installer"
+
+    def validate(self) -> None:
```

Comment:
The validate is not called in the transformer flow. If you need it to be called in the base PackageInstaller, it needs to implement `_internal_run` in this class. You can think about wrap two methods for derived class, like `_validate_package` and `_install_package`. So the loop, reboot logic are implemented in the base class.

Reply:
Sure, this is a good idea. It will need change in the schema of the base transformer as well. I somehow missed to call `validate` in the `_internal_run` method.